crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -3206,9 +3203,8 @@ impl<'db> TypeInferenceBuilder<'db> {
                     }
                 };
 
-                if let Symbol::Type(class_member, _) =
-                    class.class_member(self.db(), unary_dunder_method)
-                {
+                let meta = operand_type.to_meta_type(self.db());
+                if let Symbol::Type(class_member, _) = meta.member(self.db(), unary_dunder_method) {
                     let call = class_member.call(self.db(), &[operand_type]);
```

Comment:
Would using `Type::call_dunder` simplify this even a bit more?

Reply:
Oh hey look at that!  Yes it would